packages/front-end/components/CustomFields/CustomFields.tsx

```diff
@@ -0,0 +1,210 @@
+import {
+  closestCenter,
+  DndContext,
+  DragOverlay,
+  KeyboardSensor,
+  MouseSensor,
+  TouchSensor,
+  useSensor,
+  useSensors,
+} from "@dnd-kit/core";
+import {
+  arrayMove,
+  SortableContext,
+  verticalListSortingStrategy,
+} from "@dnd-kit/sortable";
+import usePermissions from "@/hooks/usePermissions";
+import { useAuth } from "@/services/auth";
+import track from "@/services/track";
+import { GBEdit } from "@/components/Icons";
+import {
+  SortableCustomFieldRow,
+  StaticCustomFieldRow,
+} from "@/components/CustomFields/SortableCustomField";
+import CustomFieldModal from "@/components/CustomFields/CustomFieldModal";
+import { useDefinitions } from "@/services/DefinitionsContext";
+
+const CustomFields: FC<{
+  section: CustomFieldSection;
+  title: string;
+}> = ({ section, title }) => {
+  const [modalOpen, setModalOpen] = useState<Partial<CustomField> | null>(null);
+  const permissions = usePermissions();
+  const { customFields, mutateDefinitions } = useDefinitions();
+  const { apiCall } = useAuth();
+  const [activeId, setActiveId] = useState();
+  const [items, setItems] = useState(
+    customFields?.filter((x) => x.section === section)
+  );
+
+  useEffect(() => {
+    setItems(customFields?.filter((x) => x.section === section));
+  }, [customFields, section]);
+
+  const selectedRow = useMemo(() => {
+    if (!activeId) {
+      return null;
+    }
+    return items?.find((cf) => cf.id === activeId);
+  }, [activeId, items]);
+
+  const deleteCustomField = async (customField: CustomField) => {
+    await apiCall(`/custom-fields/${customField.id}`, {
+      method: "DELETE",
+    }).then(() => {
+      track("Delete Custom Experiment Field", {
+        type: customField.type,
+      });
+      mutateDefinitions();
+    });
+  };
+
+  const sensors = useSensors(
+    useSensor(MouseSensor, {}),
+    useSensor(TouchSensor, {}),
+    useSensor(KeyboardSensor, {})
+  );
+
+  function handleDragStart(event) {
+    setActiveId(event.active.id);
+  }
+
+  async function handleDragEnd(event) {
+    if (items) {
+      const { active, over } = event;
+      if (active.id !== over.id) {
+        // it moved:
+        const oldIndex = items?.findIndex((x) => x.id === active.id);
+        const newIndex = items?.findIndex((x) => x.id === over.id);
+        const newItems = arrayMove(items, oldIndex, newIndex);
+        setItems(newItems);
+        await apiCall(`/custom-fields/reorder`, {
+          method: "POST",
+          body: JSON.stringify({
+            oldId: active.id,
+            newId: over.id,
+          }),
+        }).then(() => {
+          mutateDefinitions();
+        });
+      }
+
+      setActiveId(undefined);
+    }
+  }
+
+  function handleDragCancel() {
+    setActiveId(undefined);
+  }
+
+  return (
+    <>
+      <div className="mb-5">
+        <div className="row mb-3 align-items-center">
+          <div className="col-auto">
+            <h3>{title}</h3>
```

Comment:
Should probably be <div className="h4"> to match the other elements on the page (description, hypothesis)

Reply:
This is on the custom field listing page...